### ISSUE TITLE
[FEATURE] install csi driver as  privileged policy is disabled in k8s env

### ISSUE THREAD

**Issue by BUPTAnderson** [juicedata/juicefs-csi-driver#597]
**What feature you'd like to add:**
support install `CSI Controller` and `CSI Node (DaemonSet)` with `privileged: false`
**Why is this feature needed:**
For security reasons,  privileged feature is disabled in production environments(`privileged: false`) in some k8s production environments.
In these environments, we are also eager to use JuiceFS.

**Comment by BUPTAnderson** [comment]
@zwwhdls

**Comment by fungaren** [comment]
If you can do `mount` without `privileged: true`, then it's ok. See: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-capabilities-for-a-container

For `mount()/umount()`, `CAP_SYS_ADMIN` is required, as described in https://github.com/torvalds/linux/blob/master/include/uapi/linux/capability.h

```yaml
    securityContext:
      capabilities:
        add: ["CAP_SYS_ADMIN"]
```

**Comment by zwwhdls** [comment]
Hi @BUPTAnderson , @fungaren ,

As far as we know, the following things should be considered:

1. unprivileged pod need these permissions:
- `SYS_ADMIN` capability - set in securityContext of pod
- fuse device (/dev/fuse) - device plugin provides or client make it by `mknod` (need MKNOD capality)
- Several distributions enable the AppArmor module by default, such as Ubuntu and SUSE. Users should set `mount fstype=fuse.*,  capability sys_admin,   capability mknod,` in AppArmor profile using by mount pod. (Referring to https://kubernetes.io/docs/tutorials/security/apparmor/)

2. mount pod need propagate mountpoint to host, so that CSI node can bind mountpoint to application pod. That is [Bidirectional propagation](https://kubernetes.io/docs/concepts/storage/volumes/#mount-propagation) and **requires privileged pod**.


In conclusion, we can only sadly say that it is not currently possible to use mount pod via unprivileged.
